wsdl/ver10/schema/onvif.xsd

```diff
@@ -7967,6 +7967,10 @@ and sample rate. </xs:documentation>
 				</xs:annotation>
 			</xs:element>
 			<xs:element name="Extension" type="tt:RecordingJobConfigurationExtension" minOccurs="0"/>
+			<xs:element name="EventFilter" type="tt:RecordingEventFilter" minOccurs="0">
```

Comment:
In EventFilter, better to add **maxOccurs="unbounded"** for more flexibility, since we are using the event topic along with the content.

This allows client to add topic specific content filtering.  Otherwise content filtering will be generic applicable for all topics added to filter.

Reply:
From my point of view it only makes sense with content filtering. Drawback is, that it requires more resources. If we want to add this we should also add a capability with the maximum supported filter instances similar as we have for maximum pullpoints etc.

Reply:
Thought a bit about implementation aspects and figured out that there is no big difference between allowing multiple EventFilter and multiple Jobs. Hence I suggest to leave as is since clients or devices can generate multiple event recording jobs on different topic/value conditions.

With multiple jobs I see a general restriction that multiple jobs on the same recording need to share pre- and post alarm times to avoid requiring complex implementations.

Reply:
Allowing multiple filter will make it more generic because it can support both type of implementations,
1.) One recording job can be used to configure multiple filter based on max filter capability
2.) Adding more recording job in order to support multiple filter settings.  (In this case max filter capability can be set to 1)

On the other hand if we restrict with one filter element, we are left with only one option
1.) Adding more recording job in order to support multiple filter settings.

I am ok with keeping the change as it is, but would be better if we mention this recommendation of supporting multiple recording jobs to support detailed event content filtering, somewhere in recordingcontrol spec.

Reply:
My feeling is that the original design of having one topic filter and one content filter is not appropriate, because the content filter typically applies only to a subset of topics. How about restructuring the definition to allow pairs with the topic filter being mandatory and the content filter optional:

```	<xs:complexType name="RecordingEventFilter">
		<xs:sequence>
			<xs:element name="Filter" maxOccurs="unbounded">
				<xs:complexType>
					<xs:sequence>
						<xs:element name="Topic" type="xs:string">
							<xs:annotation><xs:documentation>Topic filter as defined in section 9.6.3 of the ONVIF Core Specification.</xs:documentation></xs:annotation>
						</xs:element>
						<xs:element name="Source" type="xs:string" minOccurs="0">
							<xs:annotation><xs:documentation>Optional message content filter as defined in section 9.4.4 of the ONVIF Core Specification.</xs:documentation></xs:annotation>
						</xs:element>
					</xs:sequence>
				</xs:complexType>
			</xs:element>
			<xs:element name="Before" type="xs:duration" minOccurs="0">
				<xs:annotation><xs:documentation>Optional timespan to record before the actual event condition became active.</xs:documentation></xs:annotation>
			</xs:element>
			<xs:element name="After" type="xs:duration" minOccurs="0">
				<xs:annotation><xs:documentation>Optional timespan to record after the actual event condition becomes inactive.</xs:documentation></xs:annotation>
			</xs:element>
			<xs:any namespace="##any" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>   <!-- first ONVIF then vendor -->
		</xs:sequence>
		<xs:anyAttribute processContents="lax"/>
	</xs:complexType>
```

Reply:
We've identified an additional issue with the content filter. It should only be used to filter the source. Filtering data fields would result in unbalanced property state reporting. Hence the mechanism cannot be used for defining more complex recording definitions like "record when count is greater than 4 and less than 8". If this is needed an additional data filter would need to be defined.

Reply:
Agree with your proposed change.